src/Http/Routing/src/Matching/AcceptsMatcherPolicy.cs

```diff
@@ -0,0 +1,392 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Headers;
+using Microsoft.AspNetCore.Http.Metadata;
+
+namespace Microsoft.AspNetCore.Routing.Matching;
+
+internal sealed class AcceptsMatcherPolicy : MatcherPolicy, IEndpointComparerPolicy, INodeBuilderPolicy, IEndpointSelectorPolicy
+{
+    internal const string Http415EndpointDisplayName = "415 HTTP Unsupported Media Type";
+    internal const string AnyContentType = "*/*";
+
+    // Run after HTTP methods, but before 'default'.
```

Comment:
Oh boy, GitHub does not like file scoped namespaces.

Reply:
I feel like we should undo it and use the feature enmasse.